scripts/cluster_network.py

```diff
@@ -249,13 +249,14 @@ def distribute_clusters(
             n.loads_t.p_set.mean()
             .groupby(n.loads.bus)
             .sum()
-            .groupby([n.buses.country])
+            .groupby([n.buses.country, n.buses.sub_network])
             .sum()
             .pipe(normed)
         )
-        assert len(L.index) == len(n.buses.country.unique()), (
+        countries_in_L = pd.unique(L.index.get_level_values(0))
+        assert len(countries_in_L) == len(n.buses.country.unique()), (
```

Comment:
Have you tested the different clustering? by gdp/pop/load...?

Reply:
Have done a set of quick tests on AZ. It works for kmeans + clustering by load/pop/gdp. But fails for hac complaining that `Cannot extract more clusters than samples: 5 clusters where given for a tree with 3 leaves`. Apart of that, there are some troubles for alternative clustering when calculating hydro profiles. 

Probably, it makes sense for that PR to focus on making work k-means clustering keeping track of the issues appeared for `hac` and alternative clustering. What do you think?

It would like to test also more complicated geometries. My feeling is that it would be great to use scenario management you have developed but I was not able to get it work... Would be very grateful if you could briefly explain that. Ready to add the needed steps to the documentation ;)

Reply:
I agree that alternative clustering may be secondary if the CI passes.
However before damaging hac, I think it would be good to investigate little to make it compatible.
My feeling is that little should be needed since Eur uses the subnetwork.
What do you think?

Reply:
Mmm I'm ok to keep alternative clustering as a secondary priority, but I feel that little time spent on hac and gadm clustering may be worth.
The gadm should be used by @Hazem-IEG right?
And hac may require little changes as the eur version employs the clustering by sub network.
Maybe we miss little to fix it.
What do you think?

Reply:
Will be happy to play with hac clustering :)

Not sure I get the point with gadm/alternative clustering: isn't it the same? (just to clarify)

Reply:
yeah sorry mb

Reply:
Thanks for clarification :)